patroni/ha.py

```diff
@@ -847,14 +976,25 @@ def _is_healthiest_node(self, members, check_replication_lag=True):
                         logger.warning('Primary (%s) is still alive', st.member.name)
                         return False
                     if my_wal_position < st.wal_position:
+                        nodes_ahead += 1
                         logger.info('Wal position of %s is ahead of my wal position', st.member.name)
                         # In synchronous mode the former leader might be still accessible and even be ahead of us.
                         # We should not disqualify himself from the leader race in such a situation.
                         if not self.is_synchronous_mode() or self.cluster.sync.is_empty\
```

Comment:
it maybe makes sense to add `self.cluster.sync.is_empty` condition description to the comment